### PR TITLE
Encoding properties

### PR DESCRIPTION
**Pull request by jcgeer**
Adding a new property (exporter.encoding) to allow users to select the encoding of output files, defaulting to UTF-8.

### CODE REVIEW

**Comment by jcgeer** [comment]
`FileWriter`s were replaced with `OutputStreamWriter`s since a `FileWriter` does not allow you to specify an encoding character set.

File: src/main/java/org/mitre/synthea/export/CDWExporter.java [hunk 1]
```diff
@@ -215,9 +216,9 @@ private CDWExporter() {
     }
   }
 
-  private FileWriter openFileWriter(Path outputDirectory, String filename) throws IOException {
+  private OutputStreamWriter openOutputStreamWriter(Path outputDirectory, String filename) throws IOException {
     File file = outputDirectory.resolve(filename).toFile();
-    return new FileWriter(file);
+    return new OutputStreamWriter(new FileOutputStream(file));
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/CDWExporter.java]: You don't seem to use the new charset `exporter.encoding` property here.

File: src/main/java/org/mitre/synthea/export/CPCDSExporter.java [hunk 2]
```diff
@@ -185,19 +194,26 @@ public static CPCDSExporter getInstance() {
...
       String encounterID = UUID.randomUUID().toString();
-      String payerId = encounter.claim.payer.uuid.toString();
       UUID medRecordNumber = UUID.randomUUID();
       CPCDSAttributes encounterAttributes = new CPCDSAttributes(encounter);
+      if (Boolean.parseBoolean(Config.get("exporter.cpcds.single_payer"))) {
+        payerId = "b1c428d6-4f07-31e0-90f0-68ffa6ff8c76";
+      }
+      else {
+        payerId = encounter.claim.payer.uuid.toString();
+      }
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/CPCDSExporter.java]: While interesting, this is unrelated to encoding. Also, why the particular hard-coded UUID?

File: src/main/java/org/mitre/synthea/export/CPCDSExporter.java [hunk 3]
```diff
@@ -185,19 +194,26 @@ public static CPCDSExporter getInstance() {
...
 
       for (CarePlan careplan : encounter.careplans) {
-        coverage(personID, encounterID, careplan, payerId, type, groupId, groupName);
+        coverage(personID, careplan.start, careplan.stop, payerId, type, groupId, groupName, name);
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/CPCDSExporter.java]: Again, unexpected changes unrelated to encodings.

File: src/main/java/org/mitre/synthea/export/CPCDSExporter.java [hunk 4]
```diff
@@ -261,26 +277,26 @@ private String patient(Person person, long time) throws IOException {
...
   private void coverage(String personID, String encounterID, CarePlan careplan,
       String payerId, String type, UUID groupId, String groupName) throws IOException {
-
+  
     StringBuilder s = new StringBuilder();
     s.append(personID).append(',');
     s.append(type).append(',');
 
-    if (careplan.stop != 0L) {
+    if (stop != 0L) {
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/CPCDSExporter.java]: This breaks the build. See the travis error logs.

File: src/main/java/org/mitre/synthea/export/CSVExporter.java [hunk 5]
```diff
@@ -144,25 +145,25 @@ private CSVExporter() {
...
-      allergies = new FileWriter(allergiesFile, append);
-      medications = new FileWriter(medicationsFile, append);
-      conditions = new FileWriter(conditionsFile, append);
-      careplans = new FileWriter(careplansFile, append);
-      observations = new FileWriter(observationsFile, append);
-      procedures = new FileWriter(proceduresFile, append);
-      immunizations = new FileWriter(immunizationsFile, append);
-      encounters = new FileWriter(encountersFile, append);
-      imagingStudies = new FileWriter(imagingStudiesFile, append);
+      patients = new OutputStreamWriter(new FileOutputStream(patientsFile, append));
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/CSVExporter.java]: Also looks like you are not using the new charset encoding here?

File: src/main/resources/synthea.properties [hunk 6]
```diff
@@ -10,7 +10,7 @@ exporter.years_of_history = 10
 exporter.split_records = false
 exporter.split_records.duplicate_data = false
 exporter.ccda.export = false
-exporter.fhir.export = true
+exporter.fhir.export = false
```
Comment (jawalonoski) [inline src/main/resources/synthea.properties]: Do not change default export settings, please.

**Comment by jcgeer** [comment]
> Looks like this still needs some changes... is this a Work In Progress?
> 
> The build is broken -- please see the Travis error log. I'm surprised it even compiled locally.
> 
> Also, it does not look like you are taking into account the charset encodings anywhere except CPCDS.

Work in progress, yes. Will have it cleaned up soon.

File: src/main/java/org/mitre/synthea/export/CPCDSExporter.java (hunk 2 above)
Comment (jcgeer) [reply src/main/java/org/mitre/synthea/export/CPCDSExporter.java]: You are correct; many of these changes are unrelated. Old code that snuck in from where I branched.
